samples/subsys/llext/fs_loader/makeimg.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+# Create a FAT12 disk image to flash in memory
+# This tool requires the following to be available on the host system:
+#
+# - dosfstools
+# - mtools
+
+PROGNAME=`basename $0`
+OUTPUT=$1
+INPUTS=${@:2}
+LOGICAL_SECTOR_SIZE=512
+DISK_SIZE_KB=128
+
+usage() {
+    echo -e "Usage:\n\t$PROGNAME output.img input1 [input2] [...]"
+}
+
+die() {
+    echo "$*"
```

Comment:
Hello Marc,
> Use shellcheck

Great, I did not know about this tool. I'll give it a try.
> If you're allergic to printf

I'd prefer use small efficient tools. _printf_ is a war machine but I can use it for the die() compatibility.
 > What is "done"?

If the script is executed without error, you get something like this:
`Creating empty 'foo.img' image...done`
`Creating FAT partition image...done`
`Copying input files...done`

This is for cosmetic only. I'll modify it to match with your _die()_ definition
> meaningless "failed" which is no better than silence

Totally agreed. Usually I redirect _stderr_ and I print it at the end so you get something like:

`Creating empty 'foo.img' image...failed`
`[followed by a long  and detailed error message]`
> backquotes are deprecated

From what I remember, they are POSIX compliant and $() style is bash'one. Where did you see it was deprecated?
I'll try to take more of your suggestions :)

Reply:
> It's almost never used in build systems because when building in parallel you will have unrelated statements between ... and done

I don't think every command in this small script requires logging before and after but if you insist:

```
Copying $INPUT to $OUTPUT ...
 building something else
 building another thing
... copying $INPUT to $OUTPUT done
```

Also: if you "pay" for one line in the logs then please stuff as many details in it as possible. Make it worthwhile. Parameters are especially useful.

Reply:
> stale review

I'm not completely familiar wit the PR process, does it mean you're ok with the last changes or there is something missing?
(btw, I ran _shellcheck_ and there is no warnings anymore)

Reply:
> dismissed my request for changes

There are still some of my review comments that you haven't addressed (most importantly: you should really add `set -e`) but I don't think any of them is blocking any more.

Whatever happens with this small script in `samples`, it's really a minuscule part of the pretty big PR so I'm never going to click the "approve" button for this entire PR after reviewing only that script. It would be really misleading.

Reply:
Well, if I now add `set -e`, the `die()` mechanism will be useless?!
>  pretty big PR

Yes, it was one of the first comment. I'll split it, I tried to fix all the errors but I cannot locally run the testsuite, so it is not really efficient this way.
Anyway, thx for your advises!

Reply:
> Well, if I now add set -e, the die() mechanism will be useless?!

No, see above.

I understand reading long explanations and discussions about shell scripting is absolutely not the priority in this PR so just try it for yourself, that's the fastest way:
1. add `set -e` at the top
2. inject some random failure. Corrupt some parameter, etc.
3. run the script

Oh and did I already mentioned to use `shellcheck`? It's like chatGPT for shell scripts but much smarter and better :-)